include/support.h

```diff
@@ -266,6 +266,21 @@ constexpr float coarse_cos(float x)
 	return coarse_sin(x + half_pi);
 }
 
+// Round up to the next highest power of 2.
+constexpr uint64_t next_power_of_two(uint64_t x)
+{
+	x--;
```

Comment:
Is this safe if it's fed a zero - Would we expect a zero to be returned?
(Will wrap around to max 64bit value, then get or'd with fractions of itself?)

Reply:
technically a 0 should return a 1, so we could add a `x += (x == 0)` at the end, but in practical terms, since this is getting called to size a buffer, you need 0 bytes to store 0 bytes.

Reply:
Right; I forgot it's 'next' power of two; so even if it's fed a perfect power of two - it will double it, because it's the next power of two?

Ignoring this function - would the closest power of two, in the positive direction be more what we want? Ie: 0 would get 0.  16380 would get 16384.  16384 would get 16384 (juat a question.. maybe we're going from blocks to stereo samples and need the doubling?)

Reply:
it works out to 'next power of two if not a power of 2':

```
0 -> 0
1 -> 1
2 -> 2
3 -> 4
4 -> 4
5 -> 8
6 -> 8
7 -> 8
8 -> 8
9 -> 16
10 -> 16
11 -> 16
12 -> 16
13 -> 16
14 -> 16
15 -> 16
16 -> 16
17 -> 32
```